6 Workbench Development Exercise - Traffic Light Widget/trafficLightWidget-wb/src/com/tridiumuniversity/trafficLightWidget/ui/BTrafficLightWidget.java

```diff
@@ -0,0 +1,86 @@
+package com.tridiumuniversity.trafficLightWidget.ui;
+
+import com.tridiumuniversity.trafficLightWidget.BTrafficLight;
+import com.tridiumuniversity.trafficLightWidget.BTrafficLightState;
+
+import javax.baja.gx.BColor;
+import javax.baja.gx.BEllipseGeom;
+import javax.baja.gx.BRectGeom;
+import javax.baja.gx.Graphics;
+import javax.baja.nre.annotations.NiagaraType;
+import javax.baja.ui.BWidget;
+import javax.baja.sys.Sys;
+import javax.baja.sys.Type;
+
+@NiagaraType
+public class BTrafficLightWidget extends BWidget {
+//region /*+ ------------ BEGIN BAJA AUTO GENERATED CODE ------------ +*/
+//@formatter:off
+/*@ $com.tridiumuniversity.trafficLightWidget.ui.BTrafficLightWidget(2979906276)1.0$ @*/
+/* Generated Tue Feb 27 10:19:51 CST 2024 by Slot-o-Matic (c) Tridium, Inc. 2012-2024 */
+
+  //region Type
+
+  @Override
+  public Type getType() { return TYPE; }
+  public static final Type TYPE = Sys.loadType(BTrafficLightWidget.class);
+
+  //endregion Type
+
+//@formatter:on
+//endregion /*+ ------------ END BAJA AUTO GENERATED CODE -------------- +*/
+
+	public void setComponent(BTrafficLight component) {
+		this.component = component;
+	}
+
+	@Override
+	public void paint(Graphics g) {
+		drawTrafficLight(g);
+		drawRedLight(g);
+		drawYellowLight(g);
+		drawGreenLight(g);
+	}
+
+	private void drawRedLight(Graphics g) {
+		BEllipseGeom light = BEllipseGeom.make(25, 25, lightDiameter, lightDiameter);
+		boolean isRedLightActive = component != null && component.getState().getOrdinal() == BTrafficLightState.RED;
+		g.setBrush(isRedLightActive ? activeRedColor : inactiveRedColor);
+		g.fill(light);
+	}
+
+	private void drawYellowLight(Graphics g) {
+		BEllipseGeom light = BEllipseGeom.make(25, 100, lightDiameter, lightDiameter);
+		boolean isYellowLightActive = component != null && component.getState().getOrdinal() == BTrafficLightState.YELLOW;
+		g.setBrush(isYellowLightActive ? activeYellowColor : inactiveYellowColor);
+		g.fill(light);
+	}
+
+	private void drawGreenLight(Graphics g) {
+		BEllipseGeom light = BEllipseGeom.make(25, 175, lightDiameter, lightDiameter);
+		boolean isGreenLightActive = component != null && component.getState().getOrdinal() == BTrafficLightState.GREEN;
+		g.setBrush(isGreenLightActive ? activeGreenColor : inactiveGreenColor);
+		g.fill(light);
+	}
+
+	private void drawTrafficLight(Graphics g) {
+		BRectGeom rectangle = BRectGeom.make(0, 0, rectangleWidth, rectangleHeight);
+		g.setBrush(trafficLightColor);
+		g.fill(rectangle);
+	}
+
+
+	private BTrafficLight component;
+
+	private static final BColor trafficLightColor = BColor.make("#FF222222");
```

Comment:
Since these are all `static final` constants, it's probably a good idea to use `SHOUTY_CASE`.